plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -45,7 +51,15 @@ public String source() {
 
     public static final YamlReader<String> READ_FROM_URL = config -> {
         final URL url = URI.create(config).toURL();
-        return new InputStreamReader(url.openStream(), UTF_8);
+        URLConnection connection = url.openConnection();
+        String user = System.getenv(ConfigurationAsCode.CASC_JENKINS_CONFIG_USER_ENV);
+        String password = System.getenv(ConfigurationAsCode.CASC_JENKINS_CONFIG_PASSWORD_ENV);
+        String token = System.getenv(ConfigurationAsCode.CASC_JENKINS_CONFIG_TOKEN_ENV);
+        if ((StringUtils.isNotBlank(user) && StringUtils.isNotBlank(password))
+                || StringUtils.isNotBlank(token)) {
+            connection.setRequestProperty("Authorization", getAuthString(user, password, token));
```

Comment:
GitLab support would be in question for instance
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html
Here you have to pass `Private-Token` in the header

Reply:
This is really causing us pain. Can move this forward and get a request started to add on GitLab support later? BasicAuth is pretty ubiquitous.

Reply:
I am not blocking this PR, the author who opened it does not respond to simple review nor questions.
I cannot alleviate your pain, if you want something please feel free to contribute and set your expectation accordingly.

![](https://www.bikalims.org/images/ConnitStriponFOSS.xcf)

Reply:
@casz Never saw notifications on this. Not sure why. This breakage and a couple others ended up forcing us to abandon CasC with our client. I still think this is a worthwhile project though. I'll try looking into a resolution in my spare time.

I've done enough work on internal forks of OSS projects and have led enough meat-space organizations to understand your pain. Most successful orgs/projects are held together by a very small group of dedicated individuals. Wasn't trying to imply you folks aren't working hard. Please keep in mind though that for those of us outside the project the why behind the death/stagnation of an individual PR is very opaque.

Anyways. I'll see what I can do.

Reply:
If someone is blocking feel free to contribute that's all I can say... 😄
As for breakage? I wonder what that could be.

As for this PR it can be solved by spinning up a proxy. 😓

Reply:
Hey there, I wanted to try to add the custom header feature, if you don't mind.

I was trying to run tests for this and add something like `CASC_JENKINS_CONFIG _CUSTOM_AUTH_HEADER`, but existing tests are failing to me. I'm trying to run them with `mvn test`.

```
[INFO] Running io.jenkins.plugins.casc.yaml.YamlSourceTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in io.jenkins.plugins.casc.yaml.YamlSourceTest
[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   SecretSourceResolverTest » Reactor java.lang.Error: java.lang.reflect.Invocati...
[ERROR]   PrimitiveConfiguratorTest » Reactor java.lang.Error: java.lang.reflect.Invocat...
[ERROR]   ExportTest.shouldNotExportValuesWithSecretConstructors » Reactor java.lang.Err...
[ERROR]   ExportTest.shouldNotExportValuesWithSecretFields » Reactor java.lang.Error: ja...
[ERROR]   ExportTest.shouldNotExportValuesWithSecretGetters » Reactor java.lang.Error: j...
[ERROR]   YamlExportTest.shouldDiscoverSecretsBasedOnTheAttributeType » Reactor java.lan...
[ERROR]   FailedTest.org.jvnet.hudson.test.JellyTestSuiteBuilder$JellyTestSuite » Reactor
[INFO]
[ERROR] Tests run: 21, Failures: 0, Errors: 7, Skipped: 0
```

Reply:
Also, how can one use [this](https://github.com/matankatz/configuration-as-code-plugin/blob/5bc8a7fa3b55ef59c2973e82dd43730cc1df8c97/plugin/src/test/java/io/jenkins/plugins/casc/yaml/YamlSourceTest.java#L23) to check if headers were set? Maybe creating a property on `YamlSource` with that header?